docs/core/tutorials/testing-library-with-visual-studio-code.md

```diff
@@ -0,0 +1,187 @@
+   ```
+
+1. Create a project reference to the class library project by running the following command:
+
+   ```dotnetcli
+   dotnet add StringLibraryTest/StringLibraryTest.csproj reference StringLibrary/StringLibrary.csproj
+   ```
+
+## Add and run unit test methods
+
+When Visual Studio runs a unit test, it executes each method that is marked with the <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute> attribute in a class that is marked with the  <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestClassAttribute> attribute. A test method ends when the first failure is found or when all tests contained in the method have succeeded.
+
+The most common tests call members of the <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert> class. Many assert methods include at least two parameters, one of which is the expected test result and the other of which is the actual test result. Some of the `Assert` class's most frequently called methods are shown in the following table:
+
+| Assert methods     | Function |
+| ------------------ | -------- |
+| `Assert.AreEqual`  | Verifies that two values or objects are equal. The assert fails if the values or objects aren't equal. |
+| `Assert.AreSame`   | Verifies that two object variables refer to the same object. The assert fails if the variables refer to different objects. |
+| `Assert.IsFalse`   | Verifies that a condition is `false`. The assert fails if the condition is `true`. |
+| `Assert.IsNotNull` | Verifies that an object isn't `null`. The assert fails if the object is `null`. |
+
+You can also use the <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert.ThrowsException%2A?displayProperty=nameWithType> method in a test method to indicate the type of exception it's expected to throw. The test fails if the specified exception isn't thrown.
+
+In testing the `StringLibrary.StartsWithUpper` method, you want to provide a number of strings that begin with an uppercase character. You expect the method to return `true` in these cases, so you can call the <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert.IsTrue%2A?displayProperty=nameWithType> method. Similarly, you want to provide a number of strings that begin with something other than an uppercase character. You expect the method to return `false` in these cases, so you can call the <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert.IsFalse%2A?displayProperty=nameWithType> method.
+
+Since your library method handles strings, you also want to make sure that it successfully handles an [empty string (`String.Empty`)](xref:System.String.Empty), a valid string that has no characters and whose <xref:System.String.Length> is 0, and a `null` string that hasn't been initialized. If `StartsWithUpper` is called as an extension method on a <xref:System.String> instance, it can't be passed a `null` string. However, you can also call it directly as a static method and pass a single <xref:System.String> argument.
```

Comment:
This should change. It is possible to call an extension method with null:

```csharp
string arg = null;
arg.StartsWithUpper();
```

Here's a thought:

```suggestion
Since your library method handles strings, you also want to make sure that it successfully handles an [empty string (`String.Empty`)](xref:System.String.Empty), a valid string that has no characters and whose <xref:System.String.Length> is 0, and a `null` string that hasn't been initialized. You can call `StartsWithUpper` directly as a static method and pass a single <xref:System.String> argument. Or, `StartsWithUpper` could be called as an extension method on a `string` variable assigned to `null`.
```